### PR TITLE
Docs/debug update

### PR DESCRIPTION
**Pull request by blackboxsw**
update manpage and add cli help documentation for --debug

### CODE REVIEW

**Comment by blackboxsw** [comment]
to test:
man ./ubuntu-advantage.1

and 

python3 -m uaclient.cli --help

File: ubuntu-advantage.1 [hunk 1]
```diff
@@ -75,6 +75,48 @@ Possible values are: active, inactive, or n/a
 Show version of the Ubuntu Advantage package.
 
 
+.SH CONFIGURATION
+By default, ubuntu-advantage client configuration options are read from
```
Comment (powersj) [inline ubuntu-advantage.1]: We typically say "Ubuntu Advantage" above, so let's stick with that

File: ubuntu-advantage.1 [hunk 2]
```diff
@@ -75,6 +75,48 @@ Possible values are: active, inactive, or n/a
...
+(/etc/ubuntu-advantage/uaclient.conf). If a (uaclient.conf) is present in
```
Comment (powersj) [inline ubuntu-advantage.1]: Add i.e.: `(i.e. /etc/ubuntu-advantage/uaclient.conf). `

Why do we tell them about a local configuration file? I think we should omit that.

File: ubuntu-advantage.1 [hunk 3]
```diff
@@ -75,6 +75,48 @@ Possible values are: active, inactive, or n/a
...
+the current working directory it will be used instead of the default conf file.
+
+The following configuration options are available:
+.TP
+.B
```
Comment (powersj) [inline ubuntu-advantage.1]: can you sort these the same as the config file itself?

I show:

- sso_auth_file
- contract_url
- data_dir
- log_level
- log_file

File: ubuntu-advantage.1 [hunk 4]
```diff
@@ -75,6 +75,48 @@ Possible values are: active, inactive, or n/a
...
+\fBcontract_url\fP
+The ubuntu advantage contract server URL. (\fBhttps://contacts.canonical.com\fP)
```
Comment (powersj) [inline ubuntu-advantage.1]: Ubuntu Advantage

Can we also remove the defaults from each option? One less thing to update when we change things.

File: ubuntu-advantage.1 [hunk 5]
```diff
@@ -75,6 +75,48 @@ Possible values are: active, inactive, or n/a
...
+.TP
+.B
+\fBsso_auth_url\fP
+The SSO authentication service URL. (\fBhttps://login.ubuntu.com\fP)
+.TP
+.B
+\fBdata_dir\fP
+Where ubuntu-advantage stores its data files
```
Comment (powersj) [inline ubuntu-advantage.1]: Where the Ubuntu Advantage client stores data files.

File: ubuntu-advantage.1 [hunk 6]
```diff
@@ -75,6 +75,48 @@ Possible values are: active, inactive, or n/a
...
+.TP
+.B
+\fBlog_file\fP
+The log file for ubuntu-advantage-tools (\fB/var/log/ubuntu-advantage.log\fP)
```
Comment (powersj) [inline ubuntu-advantage.1]: The log file for the Ubuntu Advantage client.

File: ubuntu-advantage.1 [hunk 7]
```diff
@@ -75,6 +75,48 @@ Possible values are: active, inactive, or n/a
...
+.B
+\fBlog_level\fP
+The logging level used when writing to log_file. (\fBdebug\fP)
+
+
+.P
+Additionally, any configuration option can be overridden in the environment
+by setting an environment variable prefaced by \fBUA_<uppercase_option_name>\fP.
+
+For example, the following overrides the contract_url found in uaclient.conf:
```
Comment (powersj) [inline ubuntu-advantage.1]: can you show overriding the log level instead? It seems weird to override the contract URL

File: ubuntu-advantage.1 [hunk 8]
```diff
@@ -75,6 +75,48 @@ Possible values are: active, inactive, or n/a
...
+.PP
+.nf
+.fam C
+  UA_CONTRACT_SERVER=https://mycontracts.canonical.com ua attach
+
```
Comment (powersj) [inline ubuntu-advantage.1]: extra line here

File: ubuntu-advantage.1 [hunk 9]
```diff
@@ -75,6 +75,48 @@ Possible values are: active, inactive, or n/a
...
+.TP
+.B
+\fBlog_file\fP
+The log file for ubuntu-advantage-tools (\fB/var/log/ubuntu-advantage.log\fP)
+.TP
+.B
+\fBlog_level\fP
+The logging level used when writing to log_file. (\fBdebug\fP)
+
+
```
Comment (powersj) [inline ubuntu-advantage.1]: extra line?

File: ubuntu-advantage.1 (hunk 2 above)
Comment (blackboxsw) [reply ubuntu-advantage.1]: will omit it. We probably shouldn't tell them thx.

File: ubuntu-advantage.1 (hunk 1 above)
Comment (blackboxsw) [reply ubuntu-advantage.1]: done.

File: ubuntu-advantage.1 (hunk 7 above)
Comment (blackboxsw) [reply ubuntu-advantage.1]: funny, I had that first, but put in contract_url due to our current testing need for internal canonical. makes sense to do log_level as an example though. fixed

**Pull request merged**